plugin/trino-phoenix5/src/main/java/io/trino/plugin/phoenix5/PhoenixClient.java

```diff
@@ -293,16 +292,7 @@ public boolean supportsTopN(ConnectorSession session, JdbcTableHandle handle, Li
     @Override
     protected Optional<TopNFunction> topNFunction()
     {
-        return Optional.of((query, sortItems, limit) -> {
```

Comment:
This looks like it belongs in a different commit

Reply:
This is needed so that topN is always handled by the connector. If the connector sometimes does it, and it is not apparent from the plan, I cannot do the push down.
That said, happy to break this into a separate commit, if that is better.

Reply:
if ttps://issues.apache.org/jira/browse/PHOENIX-6436 mentioned below is resolved (and shipped), we can address the TODO. It's fine to have this as a prep commit.